adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/FlagStat.scala

```diff
@@ -93,15 +94,15 @@ object FlagStat {
         val (primaryDuplicates, secondaryDuplicates) = DuplicateMetrics(p)
         new FlagStatMetrics(1,
           primaryDuplicates, secondaryDuplicates,
-          b2i(p.getReadMapped),
-          b2i(p.getReadPaired),
-          b2i(p.getReadPaired && p.getFirstOfPair),
-          b2i(p.getReadPaired && p.getSecondOfPair),
-          b2i(p.getReadPaired && p.getProperPair),
-          b2i(p.getReadPaired && p.getReadMapped && p.getMateMapped),
-          b2i(p.getReadPaired && p.getReadMapped && !p.getMateMapped),
-          b2i(mateMappedToDiffChromosome),
-          b2i(mateMappedToDiffChromosome && p.getMapq >= 5),
+          b2i(b(p.getReadMapped)),
```

Comment:
Ah, I see. Thanks for the additional context. Flagstat is meant to be run on aligned data, not unaligned data, which explains some of the oddness. I don't think it is a bad idea to sanitize the fields though, esp. if the performance impact is small. Any chance you could do a quick head to head on an aligned BAM? Given the reasoning, I'd be OK with including the commit here, but let's not squash it down on merge.